src/firefly/js/util/WebUtil.js

```diff
@@ -453,32 +453,32 @@ export function downloadSimple(url) {
 
 
 export function parseUrl(url) {
-    const parser = new URL(url);
-    const pathAry = [];
+    const {hash, host, hostname, href, origin, pathname, port, protocol, search, searchParams, username, password} = new URL(url);
 
     // Convert query string to object map with decoded key/value pairs
     const searchObject = {};
-    parser.searchParams.forEach((val, key) => searchObject[key] = val);
-
-    // Convert path string to object
-    const paths = parser.pathname.split('/');
-    paths.forEach((v) => {
-        v.split(';').forEach( (pp, idx) => {
-            if (idx > 0) {
-                pp.split(',').forEach( (kvp) => {
-                    const kv = kvp.split('=').map((s) => s.trim()).map((s) => decodeURIComponent(s));
-                    pathAry[idx-1] = Object.assign(pathAry[idx-1] || {}, {path: v, [kv[0]]: kv[1]});
-                    }
-                );
-            }
-        });
+    searchParams.forEach((val, key) => searchObject[key] = val);
+
+    const paths = pathname.replace(/^\//, '').split('/');
+    const filename = last(paths).split(';')[0];
+
+    // Convert paths into an array of [[path, {key:val}], [path, {key:val}], ...]
+    const pathAry = [];
+    paths.forEach((v, idx) => {
+        const [p, pval=''] = v.split(';');
+        const params = pval.split(',').reduce((pv, s='') => {
+                            if (s.length > 0) {
+                                const [k,v=''] = s.split('=');
+                                pv[k.trim()] = decodeURIComponent(v.trim());
+                            }
+                            return pv;
+                        }, {});
```

Comment:
It's used in several places with additional functionalities not in `new URL`.  That's why it was converted to using `new URL` instead of `document.createElement`.
The additional code you see is for supporting the additional functionalities not done by URL.